test/src/test/java/org/apache/accumulo/test/functional/SessionBlockVerifyIT.java

```diff
@@ -0,0 +1,181 @@
+    log.info("Finished waiting");
+  }
+
+  ExecutorService service = Executors.newFixedThreadPool(10);
+
+  @Test
+  public void run() throws Exception {
+    Connector c = getConnector();
+    String tableName = getUniqueNames(1)[0];
+    c.tableOperations().create(tableName);
+
+    BatchWriter bw = c.createBatchWriter(tableName, new BatchWriterConfig());
+
+    for (int i = 0; i < 1000; i++) {
+      Mutation m = new Mutation(new Text(String.format("%08d", i)));
+      for (int j = 0; j < 3; j++)
+        m.put(new Text("cf1"), new Text("cq" + j), new Value((i + "_" + j).getBytes(UTF_8)));
+
+      bw.addMutation(m);
+    }
+
+    bw.close();
+
+    Scanner scanner = c.createScanner(tableName, new Authorizations());
+    scanner.setReadaheadThreshold(20000);
+    scanner.setRange(new Range(String.format("%08d", 0), String.format("%08d", 1000)));
+
+    // test by making a slow iterator and then a couple of fast ones.
+    // when then checking we shouldn't have any running except the slow iterator
+    IteratorSetting setting = new IteratorSetting(21, SlowIterator.class);
+    SlowIterator.setSeekSleepTime(setting, Long.MAX_VALUE);
+    SlowIterator.setSleepTime(setting, Long.MAX_VALUE);
+    scanner.addScanIterator(setting);
+
+    final Iterator<Entry<Key,Value>> slow = scanner.iterator();
+
+    final List<Future<Boolean>> callables = new ArrayList<Future<Boolean>>();
+    for (int i = 0; i < 10; i++) {
+      Future<Boolean> callable = service.submit(new Callable<Boolean>() {
+        public Boolean call() {
+          while (slow.hasNext()) {
+
+            slow.next();
+          }
+          return slow.hasNext();
+        }
+      });
+      callables.add(callable);
+    }
+
+    Thread.sleep(10000);
+    log.info("Starting");
+    // let's add more for good measure.
+    for (int i = 0; i < 2; i++) {
+      Scanner scanner2 = c.createScanner(tableName, new Authorizations());
+
+      scanner2.setRange(new Range(String.format("%08d", 0), String.format("%08d", 1000)));
+
+      scanner2.setBatchSize(1);
+      Iterator<Entry<Key,Value>> iter = scanner2.iterator();
+      sinkKeyValues(iter);
+
+    }
+
+    int sessionsFound = 0;
+    // we have configured 1 tserver, so we can grab the one and only
+    String tserver = c.instanceOperations().getTabletServers().iterator().next();
+
+    final List<ActiveScan> scans = c.instanceOperations().getActiveScans(tserver);
+
+    for (ActiveScan scan : scans) {
+      // only here to minimize chance of seeing meta extent scans
+
+      if (tableName.equals(scan.getTable()) && scan.getSsiList().size() > 0) {
+        assertEquals("Not the expected iterator", 1, scan.getSsiList().size());
+        assertTrue("Not the expected iterator", scan.getSsiList().iterator().next().contains("SlowIterator"));
+        sessionsFound++;
+      }
+
+    }
+
+    assertEquals("Must have ten sessions to ensure 3509 is resolved", 10, sessionsFound);
```

Comment:
This is a little non-intuitive at a glance. Could you reword it to more clearly state why this equality check implies that ACCUMULO-3509 is resolved (instead of letting that up to the person reading a test failure)?